packages/components/src/Modal/GenericModal/GenericModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useId, useState } from "react"
+import { createPortal } from "react-dom"
+import { Transition } from "@headlessui/react"
+import FocusLock from "react-focus-lock"
+import { warn } from "~components/Modal/util/console"
+import { ModalContext } from "./context/ModalContext"
+import styles from "./GenericModal.module.scss"
+
+export type GenericModalProps = {
+  id?: string
+  isOpen: boolean
+  children: React.ReactNode
+  focusLockDisabled?: boolean
+  onEscapeKeyup?: (event: KeyboardEvent) => void
+  onOutsideModalClick?: (event: React.MouseEvent) => void
+  onAfterLeave?: () => void
+}
+
+export const GenericModal = ({
+  id: propsId,
+  children,
+  isOpen = true,
+  focusLockDisabled,
+  onEscapeKeyup,
+  onOutsideModalClick,
+  onAfterLeave: propsOnAfterLeave,
+}: GenericModalProps): JSX.Element => {
+  const reactId = useId()
+  const id = propsId ?? reactId
+
+  const labelledByID = useId()
+  const describedByID = useId()
```

Comment:
Possibly to revisit later, but I think there's usually a11y errors when the referenced id doesn't exist (eg. we don't have the label).

Reply:
Hm yeah fix later, this is how it works currently